components/inputs/input-search.js

```diff
@@ -0,0 +1,235 @@
+import { inputStyles } from './input-styles.js';
+import { LocalizeStaticMixin } from '../../mixins/localize-static-mixin.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+class InputSearch extends LocalizeStaticMixin(RtlMixin(LitElement)) {
+
+	static get properties() {
+		return {
+			disabled: { type: Boolean },
+			label: { type: String },
+			lastSearchValue: { type: String, attribute: false },
+			maxlength: { type: Number },
+			noClear: { type: Boolean, attribute: 'no-clear' },
+			placeholder: { type: String },
+			value: { type: String },
+			_focussed: { type: Boolean, attribute: false },
+			_hovered: { type: Boolean, attribute: false }
+		};
+	}
+
+	static get styles() {
+		return [inputStyles, css`
+				:host {
+					display: inline-block;
+					position: relative;
+					width: 100%
+				}
+				:host([hidden]) {
+					display: none;
+				}
+				.d2l-input {
+					padding-right: 2.2rem;
+					overflow: hidden;
+					text-overflow: ellipsis;
+					white-space: nowrap;
+					-webkit-appearance: textfield;
+				}
+				:host([dir="rtl"]) .d2l-input {
+					padding-right: 0.75rem;
+					padding-left: 2.2rem;
+				}
+				.d2l-input.d2l-input-focus {
+					padding-right: calc(2.2rem - 1px);
+				}
+				:host([dir="rtl"]) .d2l-input:hover,
+				:host([dir="rtl"]) .d2l-input:focus,
+				:host([dir="rtl"]) .d2l-input.d2l-input-focus {
+					padding-right: calc(0.75rem - 1px);
+					padding-left: calc(2.2rem - 1px);
+				}
+				d2l-button-icon {
+					--d2l-button-icon-min-height: 1.5rem;
+					--d2l-button-icon-min-width: 1.5rem;
+					--d2l-button-icon-border-radius: 4px;
+					--d2l-button-icon-focus-box-shadow: 0 0 0 1px #ffffff, 0 0 0 3px var(--d2l-color-celestine);
+					position: absolute;
+					right: 0.3rem;
+					top: 50%;
+					transform: translateY(-50%);
+				}
+				:host([dir="rtl"]) d2l-button-icon {
+					left: 0.3rem;
+					right: auto;
+				}
+			`
+		];
+	}
+
+	static get resources() {
+		return {
+			'ar': { 'search': 'بحث', 'clear': 'مسح البحث' },
+			'en': { 'search': 'Search', 'clear': 'Clear Search' },
+			'es': { 'search': 'Buscar', 'clear': 'Borrar búsqueda' },
+			'fr': { 'search': 'Rechercher', 'clear': 'Effacer la recherche' },
+			'ja': { 'search': '検索', 'clear': '検索のクリア' },
+			'ko': { 'search': '검색', 'clear': '검색 지우기' },
+			'nl': { 'search': 'Zoeken', 'clear': 'Zoekopdracht wissen' },
+			'pt': { 'search': 'Pesquisar', 'clear': 'Limpar Pesquisa' },
+			'sv': { 'search': 'Sökning', 'clear': 'Rensa sökning' },
+			'tr': { 'search': 'Ara', 'clear': 'Aramayı Temizle' },
+			'zh': { 'search': '搜索', 'clear': '清除搜索' },
+			'zh-tw': { 'search': '搜尋', 'clear': '清除搜尋' }
+		};
+	}
+
+	constructor() {
+		super();
+		this._focussed = false;
+		this._hovered = false;
+		this._lastSearchValue = '';
+		this.disabled = false;
+		this.noClear = false;
+		this.value = '';
+	}
+
+	get lastSearchValue() { return this._lastSearchValue; }
+	set lastSearchValue(val) {}
```

Comment:
To confirm - this line is so the consumer can't set this property themselves, only retrieve it (other than by breaking convention and accessing the private function, which is bad)

Reply:
Correct. This was [previously a readonly](https://github.com/BrightspaceUI/inputs/blob/master/d2l-input-search.js#L155) property, which isn't a concept in Lit other than making it "private" like this.